clients/drcachesim/simulator/tlb.cpp

```diff
@@ -87,18 +87,18 @@ tlb_t::request(const memref_t &memref_in)
         for (way = 0; way < associativity; ++way) {
             if (get_caching_device_block(block_idx, way).tag == tag &&
                 ((tlb_entry_t &)get_caching_device_block(block_idx, way)).pid == pid) {
-                stats->access(memref, true /*hit*/);
+                stats->access(memref, true /*hit*/, TAG_INVALID);
                 if (parent != NULL)
-                    parent->get_stats()->child_access(memref, true);
+                    parent->get_stats()->child_access(memref, true, TAG_INVALID);
                 break;
             }
         }
 
         if (way == associativity) {
-            stats->access(memref, false /*miss*/);
+            stats->access(memref, false /*miss*/, TAG_INVALID);
```

Comment:
This does not look correct: we are replacing something here.  This code needs the same transformation as the cache simulator code: the tag being replaced is computed down below.

Reply:
Changed the parameter type to a pointer to the cache block.